### PR TITLE
Fix HEXDEV-683 - POJO/MOJO would fail when a per-class tree was null (nothing to split)

### PR DESCRIPTION
**Pull request by arnocandel**

### CODE REVIEW

**Comment by mmalohlava** [comment]
Argh, i was too eager, the DRF tests are failing now

**Comment by mmalohlava** [comment]
The `hex.genmodel.utils.ByteBufferWrapper` does not support null trees:

```java
   /** Read from a fixed byte[]; should not be closed. */
    public ByteBufferWrapper(byte[] buf) {
        assert buf != null : "null fed to ByteBuffer.wrap";
        _bb = ByteBuffer.wrap(buf, 0, buf.length).order(ByteOrder.nativeOrder());
    }
```

**Comment by mmalohlava** [comment]
One more quick patch, however, something is still missing since tests are still failing on
```
MOJO EasyPredict Predictions mismatch for row {C4=1.9000000000000001, C3=5.0, C5=Iris-virginica, C1=6.300000000000001, C2=2.5}
  Expected predictions: [2.0, 0.0, 0.0, 1.0]
  Actual predictions:   [2.0, 0.0, 0.0, 0.0]``` 

```
@ SharedTreeMojoModel.java:441 @ public abstract class SharedTreeMojoModel extends MojoModel {
             int k = _nclasses == 1? 0 : i + 1;
             for (int j = 0; j < _ntree_groups; j++) {
                 int itree = treeIndex(j, i);
+                // Skip all empty trees
+                if (_compressed_trees[itree] == null) continue;
                 if (_mojo_version.equals(1.0)) { //First version
                     preds[k] += scoreTree0(_compressed_trees[itree], row, _nclasses, false);
                 } else if (_mojo_version.equals(1.1)) { //Second version
```

**Comment by mmalohlava** [comment]
@arnocandel i updated PR, tests are passing locally

@st-pasha can you please look at this change to make sure it is MOJO-kosher? Thx.

File: h2o-genmodel/src/main/java/hex/genmodel/MojoReaderBackend.java [hunk 1]
```diff
@@ -10,4 +10,6 @@
   BufferedReader getTextFile(String filename) throws IOException;
 
   byte[] getBinaryFile(String filename) throws IOException;
+
+  boolean exists(String name);
```
Comment (st-pasha) [inline h2o-genmodel/src/main/java/hex/genmodel/MojoReaderBackend.java]: I'd recommend renaming the parameter to `filename`, for consistency

File: h2o-genmodel/src/main/java/hex/genmodel/algos/tree/SharedTreeMojoReader.java [hunk 2]
```diff
@@ -2,6 +2,7 @@
 
 import hex.genmodel.ModelMojoReader;
 
+import java.io.File;
```
Comment (st-pasha) [inline h2o-genmodel/src/main/java/hex/genmodel/algos/tree/SharedTreeMojoReader.java]: unnecessary import

**Pull request merged**